packages/runtime/src/templates/server.test.ts

```diff
@@ -0,0 +1,63 @@
+import { mockRequest } from 'next/dist/server/lib/mock-request'
+
+import { netlifyApiFetch, getNextServer, NextServerType } from './handlerUtils'
+import { NetlifyNextServer } from './server'
+
+const NextServer: NextServerType = getNextServer()
+
+jest.mock('./handlerUtils', () => {
+  const originalModule = jest.requireActual('./handlerUtils')
+
+  return {
+    __esModule: true,
+    ...originalModule,
+    netlifyApiFetch: jest.fn(({ payload }) => {
+      switch (payload.paths[0]) {
+        case '/getStaticProps/with-revalidate/':
+          return Promise.resolve({ code: 200, message: 'Revalidated' })
+        case '/not-a-path/':
+          return Promise.resolve({ code: 404, message: '404' })
+        default:
+          return Promise.reject(new Error('Error'))
+      }
+    }),
```

Comment:
[dust] I'm not a big fan of test fixtures having logic in them as they're much harder to reason with. Maybe look to have per-test mocking instead?